packages/core/upload/server/services/metrics.js

```diff
@@ -0,0 +1,121 @@
+      keepOnlySlashesSQLString = `REPLACE(${keepOnlySlashesSQLString}, ?, ?)`;
+      queryParams.push(String(i), '');
+    }
+
+    const knex = strapi.db.connection;
+
+    /*
+      The following query goal is to count the number of folders with depth 1, depth 2 etc.
+      The query returns :
+      [
+        { depth: 1, occurence: 4 },
+        { depth: 2, occurence: 2 },
+        { depth: 3, occurence: 5 },
+      ]
+
+      The query is built as follow:
+      1. In order to get the depth level of a folder:
+        - we take their path
+        - remove all numbers (by replacing 0123456789 by '', thus the 10 REPLACE in the query)
+        - count the remaining `/`, which correspond to their depth (by using LENGTH)
+        We now have, for each folder, its depth.
+      2. In order to get the number of folders for each depth:
+        - we group them by their depth and use COUNT(*)
+    */
+
+    const folderLevelsArray = (
+      await knex(folderTable)
+        .select(
+          knex.raw(
+            `LENGTH(${keepOnlySlashesSQLString}) AS depth, COUNT(*) AS occurence`,
+            queryParams
+          )
+        )
+        .groupBy('depth')
+    ).map(map => ({ depth: Number(map.depth), occurence: Number(map.occurence) })); // values can be strings depending on the database
+
+    let product = 0;
+    let folderNumber = 0;
+    let maxDepth = 0;
+    for (const folderLevel of folderLevelsArray) {
+      product += folderLevel.depth * folderLevel.occurence;
+      folderNumber += folderLevel.occurence;
+      if (folderLevel.depth > maxDepth) {
+        maxDepth = folderLevel.depth;
+      }
+    }
+    const averageDepth = folderNumber !== 0 ? product / folderNumber : 0;
+
+    let sumOfDeviation = 0;
+    for (const folderLevel of folderLevelsArray) {
+      sumOfDeviation += Math.abs(folderLevel.depth - averageDepth) * folderLevel.occurence;
+    }
+
+    const averageDeviationDepth = folderNumber !== 0 ? sumOfDeviation / folderNumber : 0;
+
+    // File metrics
+    const assetNumber = await strapi.entityService.count(FILE_MODEL_UID);
+
+    return {
+      assetNumber,
+      folderNumber,
+      averageDepth,
+      maxDepth,
+      averageDeviationDepth,
+    };
+  },
+
+  async sendMetrics() {
+    const metrics = await this.computeMetrics();
+    strapi.telemetry.send('didSendUploadPropertiesOnceAWeek', metrics);
+
+    const metricsInfoStored = await getMetricsStoreValue();
+    await setMetricsStoreValue({ ...metricsInfoStored, lastWeeklyUpdate: new Date().getTime() });
+  },
+
+  async ensureWeeklyStoredCronSchedule() {
+    const metricsInfoStored = await getMetricsStoreValue();
+    const { weeklySchedule: currentSchedule, lastWeeklyUpdate } = metricsInfoStored;
+
+    const now = new Date();
+    let weeklySchedule = currentSchedule;
+
+    if (!currentSchedule || !lastWeeklyUpdate || lastWeeklyUpdate + ONE_WEEK < now.getTime()) {
+      weeklySchedule = getWeeklyCronScheduleAt(add(now, { minutes: 5 }));
```

Comment:
Maybe there are more things planed for the date-fns, but would be an option to create a function like this here?
( date-fns  weights 6.61 MB unpacked )
```js
function addMinutes(date, minutes) {
    return new Date(date.getTime() + minutes*60000);
}
```

Reply:
Completely missed the comment from gustav above 😄  forget about this

Reply:
@Marc-Roig I really appreciate the comment though :) I agree it is a heavy dependency, but we do have it already installed and using it in many places and therefore it doesn't make much of a difference.